private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go

```
@@ -104,39 +104,65 @@ func (h *syncHandler) IsGitCommitSynced(ctx context.Context, module bufmoduleref
 	return res.Msg.Reference.GetVcsCommit() != nil, nil
 }
 
-func (h *syncHandler) BackfillTags(
+func (h *syncHandler) SyncModuleTags(
 	ctx context.Context,
-	module bufmoduleref.ModuleIdentity,
-	alreadySyncedHash git.Hash,
-	author git.Ident,
-	committer git.Ident,
-	tags []string,
-) (string, error) {
-	service := connectclient.Make(h.clientConfig, module.Remote(), registryv1alpha1connect.NewSyncServiceClient)
-	res, err := service.AttachGitTags(ctx, connect.NewRequest(&registryv1alpha1.AttachGitTagsRequest{
-		Owner:      module.Owner(),
-		Repository: module.Repository(),
-		Hash:       alreadySyncedHash.Hex(),
-		Author: &registryv1alpha1.GitIdentity{
-			Name:  author.Name(),
-			Email: author.Email(),
-			Time:  timestamppb.New(author.Timestamp()),
-		},
-		Committer: &registryv1alpha1.GitIdentity{
-			Name:  committer.Name(),
-			Email: committer.Email(),
-			Time:  timestamppb.New(committer.Timestamp()),
-		},
-		Tags: tags,
-	}))
-	if err != nil {
+	moduleIdentity bufmoduleref.ModuleIdentity,
+	commitTags map[git.Hash][]string,
+) error {
+	repoService := connectclient.Make(h.clientConfig, moduleIdentity.Remote(), registryv1alpha1connect.NewRepositoryServiceClient)
+	var repositoryID string
+	if repoRes, err := repoService.GetRepositoryByFullName(ctx, connect.NewRequest(&registryv1alpha1.GetRepositoryByFullNameRequest{
+		FullName: moduleIdentity.Owner() + "/" + moduleIdentity.Repository(),
+	})); err != nil {
 		if connect.CodeOf(err) == connect.CodeNotFound {
 			// Repo is not created
-			return "", bufsync.ErrModuleDoesNotExist
+			return nil
 		}
-		return "", fmt.Errorf("attach git tags to module %q: %w", module.IdentityString(), err)
+		return fmt.Errorf("get repository for module identity: %w", err)
```

This should never happen, When invoking SyncModuleTags we should have already attempted to create the repo. I'd throw the error here unconditionally.

---

You'd prefer that I return `bufsync.ErrModuleDoesNotExist`? Sure, I can do that.

---

I got rid of that error, because Syncer doesn't need it, but I returned a different one.

---

Sure, there's no need for a specific err but that's ok